types/index.d.ts

```diff
@@ -0,0 +1,208 @@
+declare module "react-sweet-state" {
+  import { ComponentType, ReactNode, ReactElement } from "react";
+
+  export type SetState<TState> = (newState: Partial<TState>) => void;
+  export type GetState<TState> = () => Readonly<TState>;
+  type StoreUnsubscribe = () => void;
+
+  type RenderPropComponent<TState, TActions> = (
+    state: TState,
+    actions: TActions
+  ) => ReactNode;
+
+  type HookReturnValue<
+    TState,
+    TActions extends Record<string, ActionThunk<TState, TActions>>
+  > = [TState, BoundActions<TState, TActions>];
+
+  type ActionThunk<
+    TState,
+    TActions extends Record<string, ActionThunk<TState, TActions>>
+  > = (...args: any[]) => Action<TState, TActions>;
+
+  export type Store<
+    TState,
+    TActions extends Record<string, ActionThunk<TState, TActions>>
+  > = {
+    key: string[];
+    initialState: TState;
+    actions: TActions;
+  };
+
+  export type StoreState<TState> = {
+    getState: GetState<TState>;
+    setState: SetState<TState>;
+    key: string[];
+    subscribe: (listener: () => void) => StoreUnsubscribe;
+    mutator: SetState<TState>;
+  };
+
+  export type StateAction<
```

Comment:
I think this is probably better named `ActionArgs` o `ActionParams` as state is already used in plenty other places. 
What do you think of the previous `bla(): Action<...> => {...}` pattern? Do you think we can still find a way to support that too? Wondering if the problem is that `type Action` returns `any` instead of inferring but not sure you can just let TS infer (could not find something like `*` in Flow)

Reply:
I renamed it. I don't get what your are referring to about `bla(): Action<...> => {...}` do you have any example?

Reply:
ah go it :) yes, we would need to specify the return type. I need to turn my brain on for this one.